### PR TITLE
Add logging for roles use

### LINKED ISSUE #1745: mnt: Add logging for admin operations
## Motivation

There are cases when something wrong in role code.

## Propolsal 

Add log.info messages before and after calling user callbacks.

### First thing

- log level INFO
- Log message have to contain execution duration 
- Logging **init** for user roles, at callback start and end
- Logging **validate config** for user roles, at callback start and end
- Logging **apply config** for user roles, at callback start and end
- Logging **stop** for user roles, at callback start and end

### PR DESCRIPTION
**Pull request by panticonur**
Add logging for in`role machinery.

Logging time spent on: `init`, `validate config`, `apply config`, `stop` roles.

I didn't forget about

- [x] Changelog

Close #1745

### CODE REVIEW

File: cartridge/roles.lua [hunk 1]
```diff
@@ -26,6 +26,18 @@ local ApplyConfigError = errors.new_class('ApplyConfigError')
 local ReloadError = errors.new_class('HotReloadError')
 local StopRoleError = errors.new_class('StopRoleError')
 
+local function external_module_log(msg)
+    log.error('======================================== ' .. msg)
+end
+
+local function in_role_log(msg)
+    log.error('---------------------------------------- ' .. msg)
```
Comment (panticonur) [inline cartridge/roles.lua:34]: Remove indent `'----------------------------------------  '` before merge.

File: cartridge/roles.lua [hunk 2]
```diff
@@ -26,6 +26,18 @@ local ApplyConfigError = errors.new_class('ApplyConfigError')
...
+end
+
+local function log_line(msg)
+    log.error('++++++++++++++++++++++++++++++++++++++++ ' .. msg)
```
Comment (panticonur) [inline cartridge/roles.lua:38]: Remove indent `'++++++++++++++++++++++++++++++++++++++++  '` before merge.

File: cartridge/roles.lua [hunk 3]
```diff
@@ -467,10 +484,14 @@ local function stop()
         if (service_registry.get(role.role_name) ~= nil)
         and (type(role.M.stop) == 'function')
         then
+            log.info('Stop "%s" role', role.role_name)
+            local start_time = fiber.time64()
             local _, err = StopRoleError:pcall(role.M.stop, opts)
             if err ~= nil then
                 log.error('%s', err)
```
Comment (filonenko-mikhail) [inline cartridge/roles.lua:491]: Let's log.info with elapsed time also in error branch as additional log

File: cartridge/roles.lua [hunk 4]
```diff
@@ -467,10 +484,14 @@ local function stop()
         if (service_registry.get(role.role_name) ~= nil)
         and (type(role.M.stop) == 'function')
         then
+            log.info('Stop "%s" role', role.role_name)
+            local start_time = fiber.time64()
```
Comment (filonenko-mikhail) [inline cartridge/roles.lua:488]: Let's use clock.monotonic64 instead of fiber.time64, because fiber.time64 is the same until yield or log happens.

**Comment by yngvar-antonsson** [comment]
It doesn't cover case when apply_config is called from failover: see https://github.com/tarantool/cartridge/blob/e7cf095b0b234e9813ced34dbc8a4d342be137ad/cartridge/failover.lua#L259

**Comment by yngvar-antonsson** [comment]
Also, please add screenshot with logs as a comment in PR

**Comment by filonenko-mikhail** [comment]
> It doesn't cover case when apply_config is called from failover: see
> 
> https://github.com/tarantool/cartridge/blob/e7cf095b0b234e9813ced34dbc8a4d342be137ad/cartridge/failover.lua#L259

Very thanks.
@panticonur also use message with postfix something like (apply config from failover) to easy find apply config trigger source.

**Review by filonenko-mikhail — APPROVED** [review]

**Review by yngvar-antonsson — APPROVED** [review]

**Pull request merged**